internal/character/hook/skill.go

```diff
@@ -0,0 +1,125 @@
+package hook
+
+import (
+	"github.com/simimpact/srsim/internal/global/common"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const Skill key.Attack = "hook-skill"
+const EnhancedSkill key.Attack = "hook-enhanced-skill"
+
+func (c *char) Skill(target key.TargetID, state info.ActionState) {
+	if c.engine.HasModifier(c.id, SkillEnhancement) {
+		c.EnhancedSkill(target)
+	} else {
+		c.NormalSkill(target)
+	}
+
+	c.engine.EndAttack()
+
+	c.applySkillBurn([]key.TargetID{target})
+}
+
+func (c *char) NormalSkill(target key.TargetID) {
+	c.engine.Attack(info.Attack{
+		Key:        Skill,
+		Targets:    []key.TargetID{target},
+		Source:     c.id,
+		AttackType: model.AttackType_SKILL,
+		DamageType: model.DamageType_FIRE,
+		BaseDamage: info.DamageMap{
+			model.DamageFormula_BY_ATK: skillNormal[c.info.SkillLevelIndex()],
+		},
+		EnergyGain:   30,
+		StanceDamage: 60,
+	})
+
+	if c.engine.HasBehaviorFlag(target, model.BehaviorFlag_STAT_DOT_BURN) {
+		c.talentProc(target)
+	}
```

Comment:
ok, so about `EndAttack`, because the amount of hook's skill hits depends on various things like : 
- normal/enhanced skill
- talent is procced or not

we need to put `EndAttack` after the last hit inside every one of the above logics.
That is if we want to follow closely to the DM. I don't actually know how much of an effect this is, but the logs should at least get tidier. 
The first one is in an else here :
```suggestion
	} else {
	    c.engine.EndAttack()
	}
```
On normal skill and no talent proc, hook only hits once and the last hit is the above `Attack`.
Note : we don't want to just add `EndAttack` after normal/enhanced attack because there are a few logics after the attack ends.